packages/trip-details/src/fare-table.tsx

```diff
@@ -0,0 +1,219 @@
+import { Leg, Money } from "@opentripplanner/types";
+import flatten from "flat";
+import React from "react";
+import styled from "styled-components";
+import { FormattedMessage } from "react-intl";
+import { Transfer } from "@styled-icons/boxicons-regular/Transfer";
+
+import { boldText, renderFare } from "./utils";
+
+import { FareTableLayout, FareDetailsProps, FareTableText } from "./types";
+
+// Load the default messages.
+import defaultEnglishMessages from "../i18n/en-US.yml";
+
+// HACK: We should flatten the messages loaded above because
+// the YAML loaders behave differently between webpack and our version of jest:
+// - the yaml loader for webpack returns a nested object,
+// - the yaml loader for jest returns messages with flattened ids.
+const defaultMessages: Record<string, string> = flatten(defaultEnglishMessages);
+
+type LegAndFare = Leg & {
+  fares: Record<string, { price: Money; isTransfer?: boolean }>;
+};
+
+interface FareTypeTableProps extends FareTableLayout {
+  legs: LegAndFare[];
+  fareTotals: Record<string, Money>;
+}
+
+const TableHeader = styled.thead`
+  th {
+    min-width: 5ch;
+    padding: 0.75em 1.5em;
+    text-align: center;
+    font-weight: normal;
+  }
+  th:nth-of-type(2n + 1) {
+    background: #cccccc22;
```

Comment:
Going to leave the transparency questions to @miles-grant-ibigroup since he did that, but I'm guessing the idea is to create a darker background even on gray backgrounds (which you can see in the trip-details stories).